lib/binding_web/tree-sitter-web.d.ts

```diff
@@ -131,8 +132,28 @@ declare module 'web-tree-sitter' {
       readonly version: number;
       readonly fieldCount: number;
 
-      fieldNameForId(fieldId: number): string | null
-      fieldIdForName(fieldName: string): number | null
+      fieldNameForId(fieldId: number): string | null;
+      fieldIdForName(fieldName: string): number | null;
+      query(source: string): Query;
+    }
+
+    interface QueryResult {
+      pattern: number;
+      captures: { name: string; node: SyntaxNode }[];
+    }
+
+    interface PredicateResult {
+      operator: string;
+      operands: { name: string; type: string }[];
+    }
+
+    class Query {
+      captureNames: string[];
+
+      delete(): void;
+      matches(node: SyntaxNode, startPosition?: Point, endPosition?: Point): QueryResult[];
+      captures(node: SyntaxNode, startPosition?: Point, endPosition?: Point): QueryResult[];
```

Comment:
The `captures` and `matches` methods return objects of different types. I would call one of them `QueryCapture`, and one of them `QueryMatch`.

A `QueryCapture` [has a `node` and a `name`](https://github.com/tree-sitter/tree-sitter/blob/99cd283e39d8dfb766fb365262fd08a419dd20a2/lib/binding_web/binding.js#L1044). A `QueryMatch` (as you wrote), has a `pattern` index, and a `captures` array, whose elements are `QueryCapture` objects.

Reply:
@maxbrunsfeld updated